tdrs-backend/tdpservice/data_files/models.py

```diff
@@ -152,6 +152,20 @@ class Meta:
                                         null=True
                                         )
 
+    @property
+    def prog_type(self):
```

Comment:
not really specific to this PR, but similar logic is repeated everywhere and i keep thinking we should have a first-class member for the program type. similar to my other comment this would also allow you to use a native django list filter for the program type without creating a custom class

Reply:
Having prog_type as property means calculation is done in backend server vs DB. While normally querying the DB is faster, there are cases where pushing the query into DB is not efficient since during writing to prog_type need a db_lock. An example is calculated values where property can change base on two, three different fields.
Having said that, in this case since prog_type only needs to be updated during the initial write, I agree with your suggestion